src/lib/optimizer/strategy/join_predicate_ordering_rule.hpp

```diff
@@ -15,6 +15,9 @@ namespace opossum {
  * As such, a good predicate order is even more important that it is for regular (i.e., non-join) predicates.
  * Furthermore, the hash join only supports equals predicates, so the most selective equals predicate is moved to the
  * front.
+ *
+ * For inner joins, this is already done in AbstractJoinOrderingAlgorithm::_add_join_to_plan. See the comment over there
+ * for why we have that duplication.
  */
 class JoinPredicateOrderingRule : public AbstractRule {
  public:
```

Comment:
Consider adding some doc to `AbstractJoinOrderingAlgorithm`'s header, describing its purpose. 

Also, maybe add some sentences to describe both functions `_add_predicates_to_plan` and `_add_join_to_plan`. At least for me, unfamiliar with the algorithms, it was not very self-explanatory.

Reply:
> I wonder, whether the following comment is still relevant?

Good find! It isn't.

> Consider adding some doc to AbstractJoinOrderingAlgorithm's header, describing its purpose.

Yepp. I also just found that `AbstractJoinOrderingAlgorithm as a class is entirely pointless, because the algorithm can never be called. Will move `operator()` into the abstract method and use the doc of the greedy algorithm.

Reply:
Is this better?

Reply:
Way better :)